src/devices/src/virtio/mmio.rs

```diff
@@ -336,10 +372,18 @@ impl BusDevice for MmioDevice {
                 let v = match offset {
                     0x0 => MMIO_MAGIC_VALUE,
                     0x04 => MMIO_VERSION,
-                    0x08 => self.device.device_type(),
+                    0x08 => self
+                        .device
+                        .lock()
+                        .expect("Poisoned device lock")
```

Comment:
nit: maybe define a global `POISONED_DEV_LOCK_MSG: &str = "Poisoned device lock"` instead of copying this text all over.